third_party_integration/huggingface_transformers/BootstrapNAS_HF/0001-Modifications-for-BootstrapNAS-usage.patch

```diff
@@ -0,0 +1,1582 @@
++            logger.info("Best config: {best_config}".format(best_config=best_config))
++            logger.info("Performance metrics: {performance_metrics}".format(performance_metrics=performance_metrics))
++
++            if training_args.local_rank in [-1, 0]:
++                search_algo.visualize_search_progression()
++                search_algo.search_progression_to_csv()
++                # Best found subnet
++                elasticity_ctrl.multi_elasticity_handler.activate_subnet_for_config(best_config)
++                elasticity_ctrl.export_model(os.path.join(training_args.output_dir, "best_subnet.onnx"))
++
+     # Evaluation
+     if training_args.do_eval:
+         logger.info("*** Evaluate ***")
+diff --git a/nncf_bootstrapnas_config/nncf_bootstrapnas_bert_config_conll.json b/nncf_bootstrapnas_config/nncf_bootstrapnas_bert_config_conll.json
+new file mode 100644
+index 000000000..4e9e3a788
+--- /dev/null
++++ b/nncf_bootstrapnas_config/nncf_bootstrapnas_bert_config_conll.json
+@@ -0,0 +1,65 @@
++{
++    "input_info": [
++        {
++            "sample_size": [1, 128],
++            "type": "long"
++        },
++        {
++            "sample_size": [1, 128],
++            "type": "long"
++        },
++        {
++            "sample_size": [1, 128],
++            "type": "long"
++        }
++    ],
++    "bootstrapNAS": {
++        "training": {
++            "algorithm": "progressive_shrinking",
++            "progressivity_of_elasticity": ["depth", "width"],
++            "batchnorm_adaptation": {
++                "num_bn_adaptation_samples": 0
++            },
++            "schedule": {
++                "list_stage_descriptions": [
++                    {"train_dims": ["depth", "width"], "epochs": 3, "depth_indicator": 1, "width_indicator": 3, "init_lr": 5e-5, "epochs_lr": 3, "sample_rate": 5}
++                ]
++            },
++            "elasticity": {
++                "available_elasticity_dims": ["depth", "width"],
++                "width": {
++                    "overwrite_groups": [
++                        [
++                            "BertForTokenClassification/BertModel[bert]/BertEncoder[encoder]/ModuleList[layer]/BertLayer[0]/BertIntermediate[intermediate]/NNCFLinear[dense]/linear_0",
++                            "BertForTokenClassification/BertModel[bert]/BertEncoder[encoder]/ModuleList[layer]/BertLayer[1]/BertIntermediate[intermediate]/NNCFLinear[dense]/linear_0",
++                            "BertForTokenClassification/BertModel[bert]/BertEncoder[encoder]/ModuleList[layer]/BertLayer[2]/BertIntermediate[intermediate]/NNCFLinear[dense]/linear_0"
++                        ]
++                    ],
++                    "overwrite_groups_widths": [[3072, 2912, 2760]]
++                },
++                "depth": {
++                    "mode": "manual",
++                    "skipped_blocks": [
```

Comment:
Hope it won't be needed for NAS at some day...

Reply:
Agree. We need to increase the automation of the elasticity parameters.